compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureGenerator.java

```diff
@@ -910,7 +914,10 @@ public void visit(BLangRecordLiteral recordLiteral) {
                                                                     (BLangRecordLiteral.BLangRecordKeyValueField) field;
                 keyValueField.key.expr = rewriteExpr(keyValueField.key.expr);
                 keyValueField.valueExpr = rewriteExpr(keyValueField.valueExpr);
-            } else if (field.getKind() != NodeKind.SIMPLE_VARIABLE_REF) {
+            } else if (field.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
```

Comment:
Test for this?

Reply:
A field can be a variable reference when the static type of the expression becomes a map. right? In this case, it is not possible to write a test related to the annotation. And also this reference is to a global variable, thus it cannot also become a closure variable. 
```
const int x = 10;

function value() {
    map<int> k = {x};
}
```

Reply:
If we think we won't get here, why do we have this check.

What about

```ballerina
annotation record {| int max; |} constraints on field;

public function main() {
    int max = 1;

    record {|
        @constraints {
            max
        }
        string f;
    |} r;

    r = {f: ""};
}
```

Also this isn't just annotations, right?

Reply:
In your example, we create a `BLangRecordLiteral.BLangStructLiteral` and define its fields using key-value pairs. 
I added this check because without restricting it here, it will go to the `variable ref` visitor and check whether it is a closure variable or not. (IMO based on the current implementation, the kind of field can only be a simplevarref in the above scenario, which cannot be a closure variable anyway)

Reply:
Do we always get here after the constructs have been desugared? If so, yes, variable name fields will have been rewritten as key-value pairs, and as mentioned previously, we won't need this check.

We will need tests though, for this, spread fields, and computed keys, especially given that we may change where the closures are generated in the future.

Reply:
Any update on this?

Reply:
Yes, from the annotation desugar phase, we desugar the annotation attachments.  For now, I have removed this check

Reply:
Do we have tests with the different field kinds?

Reply:
Different field kinds are spread field, key-value field, and var named field. right? We have tests for key-value fields but cannot add an annotation for a spread field.

Reply:
https://ballerina.io/spec/lang/master/#field

> We have tests for key-value fields but cannot add annotations for a spread field.

Why do we have the subsequent check `field.getKind() != NodeKind.SIMPLE_VARIABLE_REF` then? 

```ballerina
annotation annot on field;

record {| int b; int c; |} r = {b: 1, c: 2};

int a = 0;

record {| int a; int b; int c; |} x = {a, ...<record {| @annot int b; int c; |}> r};
```

When the visit gets called here, the field kind is variable-name field.

Doesn't this visit have to be called for annotation values, btw?

Reply:
Added a test case for scenarios where a record literal is used as a default value